asset-transfer-private-data/chaincode-go/go/privateAssetTransfer.go

```diff
@@ -0,0 +1,503 @@
+	valAsbytes, err := ctx.GetStub().GetPrivateData("assetCollection", assetDeleteInput.ID) //get the asset from chaincode state
+		if err != nil {
+			return fmt.Errorf("failed to read asset: %s", err.Error())
+		}
+		if valAsbytes == nil {
+			return fmt.Errorf("asset private details does not exist: %s",  assetDeleteInput.ID)
+		}
+
+	var assetToDelete Asset
+	err = json.Unmarshal([]byte(valAsbytes), &assetToDelete)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal JSON: %s", err.Error())
+	}
+
+	// delete the asset from state
+	err = ctx.GetStub().DelPrivateData("assetCollection", assetDeleteInput.ID)
+	if err != nil {
+		return fmt.Errorf("Failed to delete state:" + err.Error())
+	}
+
+	// Also delete the asset from the color~ID index
+	indexID := "color~ID"
+	colorIDIndexKey, err := ctx.GetStub().CreateCompositeKey(indexID, []string{assetToDelete.Color, assetToDelete.ID})
+	if err != nil {
+			return err
+	}
+	err = ctx.GetStub().DelPrivateData("assetCollection", colorIDIndexKey)
+	if err != nil {
+		return fmt.Errorf("Failed to delete asset:" + err.Error())
+	}
+
+	// Finally, delete private details of asset
+
+	collection, err := getCollectionName(ctx) // get owners collection
+
+	err = ctx.GetStub().DelPrivateData(collection, assetDeleteInput.ID) // delete the asset
+	if err != nil {
+			return err
+	}
+
+	return nil
+
+}
+
+// Delete agreement to buy asset
+func (s *SmartContract) DeleteAgreement(ctx contractapi.TransactionContextInterface) error {
+
+	transMap, err := ctx.GetStub().GetTransient()
+		if err != nil {
+			return fmt.Errorf("Error getting transient: " + err.Error())
+		}
+
+	// Asset properties are private, therefore they get passed in transient field
+	transientDeleteJSON, ok := transMap["agree_delete"]
+		if !ok {
+			return fmt.Errorf("asset to delete not found in the transient map")
+		}
+
+	type assetDelete struct {
+			ID string `json:"ID"`
+		}
+
+	var assetDeleteInput assetDelete
+	err = json.Unmarshal(transientDeleteJSON, &assetDeleteInput)
+		if err != nil {
+			return fmt.Errorf("failed to unmarshal JSON: %s", err.Error())
+		}
+
+	if len(assetDeleteInput.ID) == 0 {
+		return fmt.Errorf("ID field must be a non-empty string")
+	}
+
+	// delete private details of agreement
+
+	collection, err := getCollectionName(ctx) // get owners collection
+
+	err = ctx.GetStub().DelPrivateData(collection, assetDeleteInput.ID) // delete the asset
+	if err != nil {
+			return err
+	}
+
+	// delete agreement record
+
+	assetAgreeKey, err := ctx.GetStub().CreateCompositeKey("agreeToPurchase",[]string{assetDeleteInput.ID}) // create composite key
+		if err != nil {
+			return fmt.Errorf("failed to create composite key: %s", err.Error())
+		}
+
+	err = ctx.GetStub().DelState(assetAgreeKey) // remove agreement from state
+		if err != nil {
+			return err
+		}
+
+	return nil
+
+}
+
+// Return the ID of submitting client identity. This will be used to identify
+// and verify the owner on chain
```

Comment:
Actually doesn't verify owner... just returns client id.

Reply:
this function will be removed